dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyPath.java

```diff
@@ -16,35 +16,52 @@
 
 package com.google.cloud.tools.opensource.dependencies;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
-import java.util.stream.Collectors;
 import org.eclipse.aether.artifact.Artifact;
 
-/**
- * A representation of a dependency path from a root Artifact to a node Artifact.
- */
+/** Path from the root to a node Artifact in a dependency tree. */
 public final class DependencyPath {
 
-  private List<Artifact> path = new ArrayList<>();
+  private List<Node> path = new ArrayList<>();
 
   @VisibleForTesting
-  public void add(Artifact artifact) {
-    path.add(artifact);
+  public void add(Artifact artifact, String scope, Boolean optional) {
+    path.add(new Node(artifact, scope, optional));
   }
-  
+
+  public int size() {
```

Comment:
do these new methods all need to be public? If so, they should probably have JavaDoc.

Reply:
They are not new. Size, getLeaf, getPath, and get just moved from the lower part. (getPath is renamed to getArtifacts)

Added Javadoc of these functions.